app/src/main/java/com/example/egida/presentation/viewModel/MainViewModel.kt

```diff
@@ -37,11 +47,27 @@ class MainViewModel : ViewModel() {
     }
 
     fun getUser() {
-        userDbUseCase.getUser()
-        dayUseCase.getDay()
+        viewModelScope.launch {
+            withContext(Dispatchers.Main) {
+                userDbUseCase.getUser()
+                dayUseCase.getDay()
+            }
+        }
+
+    }
+
+    fun closeDrawer() {
+
+        val currentActivity = activity
+        if (currentActivity is DrawerController) {
```

Comment:
Here you should pass a param to fragment and inside of fragment do this check and interface call.